server/src/main/java/org/apache/druid/segment/metadata/CoordinatorSegmentMetadataCache.java

```diff
@@ -502,13 +508,17 @@ public void refresh(final Set<SegmentId> segmentsToRefresh, final Set<String> da
       dataSourcesNeedingRebuild.clear();
     }
 
-    log.debug("Datasources to rebuild are [%s]", dataSourcesToRebuild);
+    log.debug("Re-building schema for datasources[%s].", dataSourcesToRebuild);
     // Rebuild the datasources.
     for (String dataSource : dataSourcesToRebuild) {
       final RowSignature rowSignature = buildDataSourceRowSignature(dataSource);
 
       if (rowSignature == null) {
-        log.info("RowSignature null for dataSource [%s], implying that it no longer exists. All metadata removed.", dataSource);
+        log.info(
+            "Datasource[%s] no longer exists as its row signature is [null]."
+            + " Removing all cached metadata.",
```

Comment:
We should also mention that removing cached metadata for this data source only.

Reply:
Will include in follow up PR.